src/Engines/thinEngine.ts

```diff
@@ -415,6 +416,17 @@ export class ThinEngine {
         return this._emptyTexture3D;
     }
 
+    /**
+     * Gets the default empty 2D array texture
+     */
+    public get emptyTexture2DArray(): InternalTexture {
+        if (!this._emptyTexture2DArray) {
+            this._emptyTexture2DArray = this.createRawTexture2DArray(new Uint8Array(4), 1, 1, 1, Constants.TEXTUREFORMAT_RGBA, false, false, Constants.TEXTURE_NEAREST_SAMPLINGMODE);

```

Comment:
I am wondering how this works as the depth should probably related to the usage ? keeping the depth at 1 might break on the shader side, should it not just wait for readiness in case of array ?

Reply:
Actually it might clamp in the shader to the actual depth so it might be ok but worth trying in WebGL :-)

Reply:
Yes it's OK, the depth coord is indeed clamped to [`max(0, min(d​-1, floor(coord​+0.5)) )`](https://www.khronos.org/opengl/wiki/Array_Texture#Access_in_shaders).